### ISSUE TITLE
How can i mock CloudBlob.CopyState?

### ISSUE THREAD

**Issue by RPM1984** [Azure/azure-storage-net#692]
Hi 👋 

I've got some code that uploads a URL to a blob in the background, utilizing the following methods:
- `CloudBlob.StartCopyAsync`
- `CloudBlob.FetchAttributesAsync`
- `CloudBlob.CopyState`

I'm trying to unit test my method that uses the above, and hence mock out the calls to Azure. The first two methods are `virtual` so i can mock out fine, but the last one is a `get`:
```
//
// Summary:
//     Gets the state of the most recent or pending copy operation.
public CopyState CopyState { get; }
```

So during my test, `CopyState.Status` will never be complete and will always be in it's initial state.

Other than creating a wrapper class (which i always try and avoid), is there any other way i can mock out this method? For example: is there a way i can set some kind of state/property in the `CloudBlob` class so that the `CopyState.Status` will have a value of `Success`?

Thanks!

**Comment by PureKrome** [comment]
@erezvani1529 Heya 👋 Any chance this issue could be scheduled for resolution in a next-release plz?

**Comment by Jordank321** [comment]
I am also waiting for this to be sorted

**Comment by BlackbaudDerek** [comment]
It looks like the ICloudBlob interface has this property, but that interface is not implemented on the CloudBlob class, itself. This is a very painful area to mock with many mocking frameworks.

**Comment by seanmcc-msft** [comment]
Fixed in version 10.0.3.

Commit: https://github.com/Azure/azure-storage-net/commit/2b863c7e0d7d586e3c978576bb74c5125d70d059

**Comment by PureKrome** [comment]
Thanks @seanmcc-msft !!!
